### PR TITLE
44 implementation of data set

### PR DESCRIPTION
**Pull request by Shtramak**
connect to #44

### CODE REVIEW

**Review by adaleorion — APPROVED** [review]

File: src/test/java/juja/sqlcmd/DatabaseManagerTest.java [hunk 1]
```diff
@@ -68,18 +78,55 @@ public void connectWhenInvalidUserPasswordReturnsFalse() {
...
         String[] expected = new String[]{"table1", "table2"};
         assertArrayEquals(expected, databaseManager.getTableNames());
     }
 
+    @Test
+    public void getTableDataWhenEmptyTableReturnsEmptyArray() throws SQLException {
+        databaseManager.connect(TEST_DB_NAME, DB_USER, DB_USER_PASSWORD);
+        executeSqlQuery("CREATE TABLE test_table()");
+        DataSet[] expected = new DataSet[]{};
+        assertArrayEquals(expected, databaseManager.getTableData("test_table()"));
```
Comment (kortov) [inline src/test/java/juja/sqlcmd/DatabaseManagerTest.java]: Change to "test_table" later

File: src/main/java/juja/sqlcmd/DataSet.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            String message = String.format("Column index should be between 0 and %s-1", size);
+            throw new IllegalArgumentException(message);
+        }
+        tableRow[columnIndex] = value;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof DataSet)) return false;
```
Comment (kortov) [inline src/main/java/juja/sqlcmd/DataSet.java]: May be if `(o == null || getClass() != o.getClass()) return false;` is better?

**Review by kortov — APPROVED** [review]
LGTM

File: src/test/java/juja/sqlcmd/DatabaseManagerTest.java (hunk 1 above)
Comment (Shtramak) [reply src/test/java/juja/sqlcmd/DatabaseManagerTest.java]: You're right. Thanks!

File: src/main/java/juja/sqlcmd/DataSet.java (hunk 2 above)
Comment (Shtramak) [reply src/main/java/juja/sqlcmd/DataSet.java]: Sure, instanceof is redutant. Thank you!

File: src/main/java/juja/sqlcmd/DatabaseManager.java [hunk 3]
```diff
@@ -35,13 +36,52 @@ public boolean connect(String database, String user, String password) {
...
+                tableRow.insertValue(i, columnValue);
+            }
+            tableData[rowNumber++] = tableRow;
+        }
+        return tableData;
+    }
+
     private int numberOfEntries(String tableName) throws SQLException {
-        String sqlQuery = "SELECT COUNT(*) as RECORDS FROM " + tableName;
+        String sqlQuery = String.format("SELECT COUNT(*) as RECORDS FROM %s", tableName);
```
Comment (kortov) [inline src/main/java/juja/sqlcmd/DatabaseManager.java]: Methods should be verbs https://goo.gl/eAg3Gg 
May be getNumberOfEntries?

File: src/main/java/juja/sqlcmd/DatabaseManager.java [hunk 4]
```diff
@@ -35,13 +36,52 @@ public boolean connect(String database, String user, String password) {
...
+            int rowSize = rsmd.getColumnCount();
+            return tableRows(resultSet, rowSize, tableSize);
+        }
+    }
+
+    public void closeConnection() throws SQLException {
+        if (connection != null) connection.close();
+    }
+
+    private DataSet[] tableRows(ResultSet resultSet, int rowSize, int tableSize) throws SQLException {
```
Comment (kortov) [inline src/main/java/juja/sqlcmd/DatabaseManager.java]: Methods should be verbs https://goo.gl/eAg3Gg
May be getTableRows?

**Review by ivshapovalov — APPROVED** [review]

**Pull request merged**